app/controllers/document_lead_image_controller.rb

```diff
@@ -117,6 +117,30 @@ def choose_image
     redirect_to document_path(document)
   end
 
+  def delete_image
+    document = Document.find_by_param(params[:document_id])
+    image = document.images.find(params[:image_id])
+
+    begin
+      if image.id == document.lead_image_id
```

Comment:
I notice you've checked by id here and by object equivalence in https://github.com/alphagov/content-publisher/pull/344/commits/4a9c904b24b7622a94db6b17cb6aa0ac3c085d0e#diff-da8c1e6384debf22b83216459a49c33fR21 any reason for discrepancy? do we save a query at all checking by id

Reply:
Ah, good spot. I was being consistent with what was in the controller. In fact, the call in the view is worse because it does an unnecessary DB query, so I changed that instead :-).